### PR TITLE
Add typescript types and onboarding ping

### PR DESCRIPTION
**Pull request by Hamish-taylor**

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+
+    /**
+     * Prevent uncaught errors from being sent.
+     */
+    disableErrorTracking?: boolean | undefined;
+
+    /**
+     * Prevent Pulse real user monitoring events from being sent.
+     */
+    disablePulse?: boolean | undefined;
```
Comment (Olwiba) [inline types/index.d.ts:55]: these types need to be updated before we can look to ship this one 👌

**Review by Olwiba — CHANGES_REQUESTED** [review]
I'm happy with the code changes here, we just need to update the type definitions to match before we can look to ship this. Nice one 🚀

**Review by Olwiba — APPROVED** [review]
Happy with the changes 🚀

**Review by PanosNB — CHANGES_REQUESTED** [review]
Primary concern is we need to get the tests to pass before releasing anything. Also check some minor typos and questions please: :-)

File: src/raygun.loader.js [hunk 2]
```diff
@@ -218,6 +229,41 @@
...
+        },
+        body: JSON.stringify(data)
+    }).then(function(response) {
+    if (response.ok) {
+      metadata.ping.failedPings = 0;
+    } else {
+      // Request failed
+      metadata.ping.failedPings++;
+    }
+  })
```
Comment (PanosNB) [inline src/raygun.loader.js:260]: This is a bit ugly here, we normally put the .catch in the line. No worries for now

**Review by PanosNB — DISMISSED** [review]
Ok for prerelease and thus, beta-testing on our systems

**Review by PanosNB — APPROVED** [review]
Again, all good for prerelease and soaking in our systems

File: CHANGELOG.md [hunk 3]
```diff
@@ -28,6 +28,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Unused normalize.css file
   
 -->
+## [3.1.1]
```
Comment (phillip-haydon) [inline CHANGELOG.md:31]: Should be 3.1.0

File: CHANGELOG.md [hunk 4]
```diff
@@ -28,6 +28,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Added
```
Comment (phillip-haydon) [inline CHANGELOG.md:33]: Add a change log to note we have new alias option for enableRealUserMonitoring

**Review by phillip-haydon — CHANGES_REQUESTED** [review]
Couple of comments.

File: bower.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "raygun4js",
-  "version": "2.28.0",
+  "version": "3.1.1",
```
Comment (CmdrKeen) [inline bower.json:3]: @Hamish-taylor Update this please (as you suggested)

File: package.json [hunk 6]
```diff
@@ -2,11 +2,12 @@
...
   "main": "dist/raygun.umd.js",
   "files": [
-    "dist/*"
+    "dist/*",
+    "types/*"
   ],
   "title": "Raygun4js",
   "description": "Raygun.com plugin for JavaScript",
-  "version": "3.0.1",
+  "version": "3.1.1",
```
Comment (CmdrKeen) [inline package.json:10]: @Hamish-taylor Update here too :)

**Review by CmdrKeen — DISMISSED** [review]
Looks good to me @Hamish-taylor - just updates on the version numbers.

**Review by CmdrKeen — DISMISSED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -28,6 +28,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Unused normalize.css file
   
 -->
+## [3.1.0]
+
+### Added 
+- Added a ping to tell Raygun that raygun4js has been set up correctly, this information will be used to help users when setting up a new application. This can be disabled using the new `sendPing` option e.g., `rg4js('sendPing', false)`. The ping contains the version of Raygun4js being used, aswell as what products that have been enabled (CrashReporting and or Real User Monitoring) 
+- Moved the typescript types into the main raygun4js repo
+- Added an new option for enabling Real user monitoring
```
Comment (JasenPalmer) [inline CHANGELOG.md:36]:
```suggestion
- Added a ping to tell Raygun that raygun4js has been set up correctly. The ping is enabled by default but can be disabled using the new `sendPing` option e.g., `rg4js('sendPing', false)`. The ping contains the version of Raygun4js and which products have been enabled (CrashReporting and/or Real User Monitoring). This information will be used to help users when setting up a new application.
- Moved the typescript types into the main raygun4js repo
- Added a new option for enabling Real User Monitoring. `rg4js('enableRealUserMonitoring', true);`
```

**Review by PanosNB — APPROVED** [review]
Ok, follow the usual prelease/soaking process please

**Review by phillip-haydon — APPROVED** [review]

**Pull request merged**